spyder/app/mainwindow.py

```diff
@@ -288,6 +288,7 @@ def signal_handler(signum, frame=None):
                 pass
         else:
             signal.signal(signal.SIGTERM, signal_handler)
+            signal.signal(signal.SIGINT, signal_handler)

```

Comment:
Please change this to

```
if not DEV:
    ...
```

Reply:
Why?

Reply:
Because it's useful to have it when working in DEV mode. That way you can get a traceback when Spyder freezes for whatever reason.

Reply:
>Because it's useful to have it when working in DEV mode. That way you can get a traceback when Spyder freezes for whatever reason.

Still this is very unconventional behavior. I am sure there must be a way where we can get a traceback AND quit...

Reply:
I think very few people uses DEV, so I don't think we should look for more.

Reply:
>Still this is very unconventional behavior. I am sure there must be a way where we can get a traceback AND quit...

Raising KeyInterruptError in the internal console when the traceback correspond to a KeyInterruptError

Reply:
could we do that? raise something, wait a bit and then close spyder?

Reply:
Hmm yes, raising a exception in the  internal console will make spyder quit, although the traceback won't correspond 😕

Reply:
>Hmm yes, raising a exception in the internal console will make spyder quit, although the traceback won't correspond 😕

Then it is not the solution I requested!

Reply:
We could also print the other traceback, although I don't like that option.

Maybe skipping the keyboard error where internal console catch errors, although I don't understand the internals of the console and idk if it's possible

Reply:
@rlaverde, please finish this one with my initial suggestion. I know that's not ideal, but it will affect mainly to us.